"Working on" commit

antariuz opened the pull request.

antariuz committed:
"Working on" commit

iddqd2d commented on src/best/com/Engine.java:
```diff
@@ -0,0 +1,9 @@
+
+    private String brand;
+    private String model;
+    private String typeOfFuel;
```
typeOfFuel поменяй тип String -> ENUM

antariuz replied:
Done!

iddqd2d commented on src/best/com/FuelTank.java:
```diff
@@ -0,0 +1,50 @@
+public class FuelTank {
+
+    private BigDecimal volume;
+    private int currentVolume;
```
оба поля BigDecimal, нет интов

antariuz replied:
Done!

iddqd2d commented on src/best/com/FuelTank.java:
```diff
@@ -0,0 +1,50 @@
+        this.currentVolume = currentVolume;
+    }
+
+    public void setWheels(List<Wheel> wheels) {
```
зачем тут колеса?

antariuz replied:
Done!

iddqd2d commented on src/best/com/Main.java:
```diff
@@ -3,6 +3,10 @@
+
+        Vehicle currentCar = new Vehicle();
+        Wheel wheels = new Wheel();
+        currentCar.addWheel(wheel);
```
создай колесо и клонируй его

antariuz replied:
Вроде разобрался с клонированием, но не понимаю, мне нужно реализовать "глубокое" клонирование, ведь так?

iddqd2d commented on src/best/com/Passenger.java:
```diff
@@ -0,0 +1,10 @@
+    private int age;
+    private String name;
+    private String surname;
+    private String sex;
```
Enum

antariuz replied:
Done!

iddqd2d commented on src/best/com/Vehicle.java:
```diff
@@ -0,0 +1,71 @@
+
+public class Vehicle {
+
+    private int passengerQuantity;
```
List<Passenger> и его size тебе в помощь не каких passengerQuantity

antariuz replied:
Done!

iddqd2d commented on src/best/com/Vehicle.java:
```diff
@@ -0,0 +1,71 @@
+    private List<Wheel> wheels;
+    private List<Door> doors;
+    private String engine;
+    private int currentFuelVolume;
```
private FuelTank fuelTank; никаких currentFuelVolume и т.п

antariuz replied:
Done!

iddqd2d commented on src/best/com/Vehicle.java:
```diff
@@ -0,0 +1,71 @@
+    private Date recentActivityDate;
+
+    public void drive(boolean checkPassenger) {
+        if (checkPassenger) {
```
проверь еще есть ли топливо в баке

antariuz replied:
Done!

iddqd2d replied:
это ошибка, значит ексепшион

iddqd2d commented on src/best/com/Door.java:
```diff
@@ -0,0 +1,7 @@
+package best.com;
+
+public class Door {
+
```
Enum тип двери: передняя, левая и т.п
методы открыть/закрыть окно

antariuz replied:
Done!

iddqd2d commented on src/best/com/Status.java:
```diff
@@ -0,0 +1,16 @@
+package best.com;
+
+public class Status {
```
че за класс такой?

antariuz replied:
Думаю там реализовать метод toString и вывод информации о машине

iddqd2d replied:
toString обьекта Vehicle должен быть в обьекте Vehicle

iddqd2d replied:
удали нафиг этот класс

iddqd2d commented on src/best/com/Vehicle.java:
```diff
@@ -0,0 +1,71 @@
+    }
+
+    public void pickUpPassenger(Passenger passenger) {
+        // pick up
```
где реализация?

antariuz replied:
Working on...

iddqd2d commented on src/best/com/FuelTank.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    public void setVolume(BigDecimal volume) {
+        if ((volume != 30) || (volume != 50) || (volume != 70)) {
```
ни каких цифр в уме, создай константы класса

antariuz replied:
Done!

iddqd2d commented on src/best/com/Vehicle.java:
```diff
@@ -0,0 +1,71 @@
+    public void drive(boolean checkPassenger) {
+        if (checkPassenger) {
+            // движение
+        } else System.out.println("В машине нет хотя бы одного водителя старше 18 лет");
```
никаких System.out.println, кидай YourException

antariuz replied:
Done!

iddqd2d commented on src/best/com/FuelTank.java:
```diff
@@ -0,0 +1,50 @@
+
+    public void setVolume(BigDecimal volume) {
+        if ((volume != 30) || (volume != 50) || (volume != 70)) {
+            System.out.println("The volume of Fuel Tank should be 30, 50 or 70 litres");
```
YourException

antariuz replied:
Done!

antariuz committed:
"Working on" commit

antariuz committed:
"Working on" commit

iddqd2d commented on src/best/com/FuelTank.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    public void checkFuelVolume() throws MyExceptions {
+        throw new MyExceptions("The fuel tank is full of " + getCurrentVolume() + " litres");
```
не надо везде пихать ексепшон, ошибка есть ошибка, она появляется в случае нарушения

iddqd2d replied:
remove this method

antariuz replied:
Done!

iddqd2d commented on src/best/com/Main.java:
```diff
@@ -2,7 +2,15 @@
+
+        Vehicle currentCar = new Vehicle();
+        Wheel wheel = new Wheel();
+        Wheel wheel_2_clone = (Wheel) wheel.clone();
```
https://khasang.io/courses/cleancode/lectures/4428018

дя названия переменных только camelCase
создай метод, который сразу отдает n-ое количество уникальных колес

iddqd2d commented on src/best/com/Wheel.java:
```diff
@@ -0,0 +1,19 @@
+        return super.clone();
+    }
+
+    public void setWheels(List<Wheel> wheels) {
```
для всех полей классОВ, которые участвуют в его состоянии должны быть геттеры и сеттеры (ALT + INSERT) -> getter and setter

iddqd2d commented on src/best/com/Passenger.java:
```diff
@@ -0,0 +1,10 @@
+
+public class Passenger {
+
+    private int age;
```
я думаю чаще всего у тебя будут обьекты вместо примитивов, сделай Integer

iddqd2d replied:
getter, setter

antariuz replied:
Done

iddqd2d commented on src/best/com/Door.java:
```diff
@@ -0,0 +1,15 @@
+
+public class Door {
+
+    private enum side {LEFT_FRONT_DOOR, RIGHT_FRONT_DOOR, LEFT_REAR_DOOR, RIGHT_REAR_DOOR};
```
правила к названиям Enum, такие как к Class

iddqd2d replied:
продолжи мысль, добавь поле класса private Side sideDoor; (getter, setter for it)
добавь поле private boolean isOpenWindows; (getter, setter for it)

iddqd2d replied:
add all args constructor for each class

iddqd2d replied:
Struct:
class {
constants;
other fields;
constructors;
getter and setters;
other methods;
}

iddqd2d commented on src/best/com/Door.java:
```diff
@@ -0,0 +1,15 @@
+
+    private enum side {LEFT_FRONT_DOOR, RIGHT_FRONT_DOOR, LEFT_REAR_DOOR, RIGHT_REAR_DOOR};
+
+    public void closeWindow(enum side) {
```
remove it

iddqd2d commented on src/best/com/Door.java:
```diff
@@ -0,0 +1,15 @@
+
+    }
+
+    public void openWindow(enum side) {
```
same

iddqd2d commented on src/best/com/Engine.java:
```diff
@@ -0,0 +1,9 @@
+
+    private String brand;
+    private String model;
+    private enum typeOfFuel {PETROL, DIESEL, GAS};
```
правила к названиям Enum, такие как к Class. Продолжи мысль...

(getter, setter for them)

iddqd2d commented on src/best/com/FuelTank.java:
```diff
@@ -0,0 +1,51 @@
+
+    public static BigDecimal FUEL_CAPACITY_30_L = new BigDecimal(30);
+    public static BigDecimal FUEL_CAPACITY_50_L = new BigDecimal(50);
+    public static BigDecimal FUEL_CAPACITY_70_L = new BigDecimal(70);
```
public? Do you use constants outside class?

iddqd2d replied:
constants must be before sample fields

iddqd2d commented on src/best/com/FuelTank.java:
```diff
@@ -0,0 +1,51 @@
+            throw new MyExceptions("The fuel tank is fully refueled");
+        } else {
+            currentVolume = currentVolume.add(inFuel);
+            throw new MyExceptions("The fuel tank is refueled to " + getCurrentVolume() + " litres");
```
remove Exception

iddqd2d commented on src/best/com/Passenger.java:
```diff
@@ -0,0 +1,10 @@
+    private int age;
+    private String name;
+    private String surname;
+    private enum sex {Male, Female};
```
continue....

iddqd2d commented on src/best/com/Vehicle.java:
```diff
@@ -0,0 +1,52 @@
+        // pick up
+    }
+
+    public void dropPassenger(Passenger passenger) {
```
when?

iddqd2d commented on src/best/com/Wheel.java:
```diff
@@ -0,0 +1,19 @@
+        return super.clone();
+    }
+
+    public void setWheels(List<Wheel> wheels) {
```
why?

iddqd2d replied:
This method must be in the Vehicle

iddqd2d commented on src/best/com/FuelTank.java:
```diff
@@ -0,0 +1,51 @@
+        return currentVolume;
+    }
+
+    public void setVolume(BigDecimal volume) throws MyExceptions {
```
what will happen if volume == null?

iddqd2d replied:
check and say me

iddqd2d commented on src/best/com/FuelTank.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    public void setCurrentVolume(BigDecimal currentVolume) {
+        this.currentVolume = currentVolume;
```
check volume capacity before set

iddqd2d commented on src/best/com/FuelTank.java:
```diff
@@ -0,0 +1,51 @@
+        this.currentVolume = currentVolume;
+    }
+
+    public void refuel(BigDecimal inFuel) throws MyExceptions {
```
remove or rename method to setCurrentVolume

iddqd2d replied:
add and use default getter and setter. ALT + Insert .

antariuz committed:
"Working on" commit

antariuz committed:
Working on Wheel class. Done:
Class Passenger
Builder patter of Passenger

antariuz committed:
Working on Wheel class. Done:
Door class
Engine class
FuelTank class
Passenger class
Wheel class
& a lot of fixes

antariuz committed:
Working on Wheel class. Done:
FuelTank class some fixes
drive
startMove
hasEnoughFuel
and etc.
